store/sqlstore/plugin_store.go

```diff
@@ -71,6 +71,43 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) store.StoreChann
 	})
 }
 
+func (ps SqlPluginStore) CompareAndUpdate(kv *model.PluginKeyValue, oldValue []byte) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		if result.Err = kv.IsValid(); result.Err != nil {
+			return
+		}
+
+		if ps.DriverName() == model.DATABASE_DRIVER_POSTGRES || ps.DriverName() == model.DATABASE_DRIVER_MYSQL {
+			updateResult, err := ps.GetMaster().Exec(
+				`UPDATE PluginKeyValueStore SET PValue = :New WHERE PluginId = :PluginId AND PKey = :Key AND PValue = :Old`,
+				map[string]interface{}{
+					"PluginId": kv.PluginId,
+					"Key":      kv.Key,
+					"Old":      oldValue,
+					"New":      kv.Value,
+				})
```

Comment:
nit: can you bring down the parenthesis here:
```suggestion
				},
			)
```